### PR TITLE
Update dependencies

### PR DESCRIPTION
**Pull request by denwilliams**
Updated all dependencies, allow prom-client v15 to be used, fix github actions

Fixed linting

### CODE REVIEW

File: .eslintrc.js [hunk 1]
```diff
@@ -9,8 +9,9 @@ module.exports = {
     "eslint:recommended",
     "plugin:ava/recommended",
     "plugin:@typescript-eslint/recommended",
-    "plugin:prettier/recommended",
-    "prettier/@typescript-eslint",
```
Comment (denwilliams) [inline .eslintrc.js:13]: These are a pain to keep up to date. Recommendation is to not use them.

Seems to work fine in vscode, and added a check to the lint script

(also we weren't even running lint on CI before!)

File: SCHEMAS.md [hunk 2]
```diff
@@ -12,9 +12,9 @@ responses validation (Via [AJV](https://ajv.js.org/json-type-definition.html)).
 
 In this document it may use language such as:
 
-- __MUST__ &rarr; this is something you always do
-- __MUST NOT__ &rarr; this is something that you should never do
-- __SHOULD__ &rarr; this is something that is a recommendation and generally is a good idea to follow to avoid issues
+- **MUST** &rarr; this is something you always do
```
Comment (denwilliams) [inline SCHEMAS.md:15]: prettier v3 side effects

File: .eslintrc.js [hunk 3]
```diff
@@ -9,8 +9,9 @@ module.exports = {
...
   ],
   ignorePatterns: ["dist/**/*"],
+  rules: {
+    "@typescript-eslint/no-explicit-any": "warn",
+  },
```
Comment (denwilliams) [inline .eslintrc.js:16]: `any` is used already in the project all over, but we werent running lint

have kept it as warnings

File: package.json [hunk 4]
```diff
@@ -8,8 +8,8 @@
     "dist"
   ],
   "scripts": {
-    "fix": "eslint . --fix",
-    "lint": "eslint .",
+    "fix": "eslint . --fix && prettier --write .",
+    "lint": "eslint . && prettier --check .",
```
Comment (denwilliams) [inline package.json:12]: run prettier check with lint step

File: .github/workflows/npm-publish.yml [hunk 5]
```diff
@@ -0,0 +1,31 @@
...
+      matrix:
+        node-version: [18.x, 20.x, 22.x]
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: ${{ matrix.node-version }}
+      - run: npm cit
+      - run: npm run lint
```
Comment (denwilliams) [inline .github/workflows/npm-publish.yml:17]: we were only running tests, run lint too

File: .github/workflows/npm-publish.yml [hunk 6]
```diff
@@ -0,0 +1,31 @@
...
+  publish:
+    needs: test
+    runs-on: ubuntu-latest
+
+    steps:
+      - uses: actions/checkout@v4
+      - uses: actions/setup-node@v4
+        with:
+          node-version: 20.x
+      - run: npm ci
```
Comment (denwilliams) [inline .github/workflows/npm-publish.yml:28]: we were actually packaging with npm install here, not ci

updated

**Review by smithamax — APPROVED** [review]

**Pull request merged**